doc/releases/release-notes-1.14.rst

```diff
@@ -36,7 +36,45 @@ Drivers and Sensors
 Networking
 **********
 
-* TBD
+* The BSD socket API should be used by applications for any network
+  connectivity needs. Old legacy net-app API is removed.
+* Majority of the network sample applications are converted to use
+  BSD socket API.
+* Implement getnameinfo(), shutdown(), freeaddrinfo(), gethostname(),
+  getsockopt(), setsockopt(), select()
+* Convert BSD socket code to use global file descriptor numbers
+* Network packet (net-pkt) overhaul. The new net-pkt API uses less memory and
+  is more streamlined.
```

Comment:
Removed old net_pkt api (https://github.com/zephyrproject-rtos/zephyr/pull/13618) not merged though. But eventually it will merge.